File: DSA Self Paced/Recursion/LectureProblems/GenerateSubsets.java
public class GenerateSubsets {
    
    static void generateSubsets(String str, int start, String current)
    {
        if (str.length() == start)
        {
            if (current == "")
                System.out.print("{Empty Set}" + " ");
            
            else
                System.out.print(current + " ");
            
            return;
        }
    
        generateSubsets(str, start+1, current);
        generateSubsets(str, start+1, current + str.charAt(start));
    }
    

    public static void main(String[] args) {
        
        String str = "ABC";
        generateSubsets(str, 0, "");
    }
}

File: DSA Self Paced/Recursion/LectureProblems/GenerateSubsets.cpp
/**
 * @file GenerateSubsets.cpp
 * @author ADITYA BHARDWAJ
 * @brief 
 * @version 0.1
 * @date 2022-08-21 10:40PM
 * 
 * @copyright Copyright (c) 2022
 * 
 */

#include <bits/stdc++.h>

using namespace std;

void generateSubsets(string &str, int start=0, string current="")
{
    if (str.length() == start)
    {
        if (current == "")
            cout << "{Empty Set}" << " ";
        
        else
            cout << current << " ";
        
        return;
    }

    generateSubsets(str, start+1, current);
    generateSubsets(str, start+1, current + str[start]);
}

int main()
{
    string str = "ABC";
    generateSubsets(str);
    return 0;
}